lib/acts_as_list/active_record/acts/no_update.rb

```diff
@@ -4,39 +4,98 @@ module List
       module NoUpdate
         extend ActiveSupport::Concern
 
+        class ArrayTypeError < SyntaxError
+          def initialize
+            super("The first argument must be an array")
+          end
+        end
+
+        class DisparityClassesError < NotImplementedError
+          def initialize
+            super("The first argument should contain ActiveRecord or ApplicationRecord classes")
+          end
+        end
+
         module ClassMethods
           # Lets you selectively disable all act_as_list database updates
           # for the duration of a block.
           #
           # ==== Examples
-          #   ActiveRecord::Acts::List.acts_as_list_no_update do
-          #     TodoList....
-          #   end
           #
-          #   TodoList.acts_as_list_no_update do
-          #     TodoList....
-          #   end
+          # class TodoList < ActiveRecord::Base
+          #   has_many :todo_items, -> { order(position: :asc) }
+          # end
+          #
+          # class TodoItem < ActiveRecord::Base
+          #   belongs_to :todo_list
+          #
+          #   acts_as_list scope: :todo_list
+          # end
+          #
+          # TodoItem.acts_as_list_no_update do
+          #   TodoList.first.update(position: 2)
+          # end
+          #
+          # Also you can pass an argument as array of extracted calsses
+          # to disable from database updates.
+          # It might be any class that is able to acts as list.
+          #
+          # ==== Examples
+          #
+          # class TodoList < ActiveRecord::Base
+          #   has_many :todo_items, -> { order(position: :asc) }
+          # end
+          #
+          # class TodoItem < ActiveRecord::Base
+          #   belongs_to :todo_list
+          #   has_many :todo_attachments, -> { order(position: :asc) }
+          #
+          #   acts_as_list scope: :todo_list
+          # end
           #
-          def acts_as_list_no_update(&block)
-            NoUpdate.apply_to(self, &block)
+          # class TodoAttachment < ActiveRecord::Base
+          #   belongs_to :todo_list
+          #   acts_as_list scope: :todo_item
+          # end
+          #
+          # TodoItem.acts_as_list_no_update([TodoAttachment]) do
+          #   TodoItem.find(10).update(position: 2)
+          #   TodoAttachment.find(10).update(position: 1)
+          #   TodoAttachment.find(11).update(position: 2)
+          # end
+
+          def acts_as_list_no_update(extra_klasses = [], &block)
```

Comment:
Since `extra_klasses` isn't a reserved word I think it should be `extra_classes`. Would you agree @swanandp?